pkg/bosh/bpmconverter/volume_factory.go

```diff
@@ -243,6 +263,32 @@ func (f *VolumeFactoryImpl) GenerateBPMDisks(manifestName string, instanceGroup
 					"ephemeral": "true",
 				},
 			}
+
+			if persistent {
```

Comment:
I think the whole func is to deeply nested. If you want, this is an opportunity to extract some blocks to methods.

```
if hasEphemeralDisk {
  disks := append(disks, newEphemeralDisk(...))
}
```